main/views.py

```diff
@@ -1,23 +1,77 @@
 from django.shortcuts import render, redirect
 from django.http import HttpResponse
 from .forms import GeolocatorStatsForm
+from .forms import GeolocatorStatsFormHistory
 from .models import GeolocatorStats
 import requests
+import re
 
+# HOW TO DEAL WITH ERROR MESSAGES 
+history = []
+#navbar, history, about, home
 def index(request):
     resp = ""
     if request.method == "POST":
         form = GeolocatorStatsForm(request.POST)
+        print(form)
         if form.is_valid():
             resp = form.cleaned_data['query']
-            form.save()
-            return render(request, 'locator.html', {"form": form, "queries": forward_geocode(resp)}) 
+            result = validate_input(resp)
+            print(result)
+            #provjeriti unos da li je koordinate ili naziv, mozda koristiti regex
+            if(result==None):
+                context = {"form": form}  
+                return render(request,'locator.html', context=context)
+            if(len(result) == 1):
+                print(resp)
+                fwg = forward_geocode(resp)
+                if(not fwg):
+                    context = {"form": form}  
+                    return render(request,'locator.html', context=context)
+                else:
+                    form.instance.query = fwg[0].get('name')
```

Comment:
We talked about this. Create 'manually' a new record in the database with all necessary attributes.